modal/_asgi.py

```diff
@@ -17,45 +16,31 @@ async def fn(scope):
         assert function_call_id, "internal error: function_call_id not set in asgi_app() scope"
 
         # TODO: Add support for the ASGI lifecycle spec.
-        messages_from_app: asyncio.Queue[List[Dict[str, Any]]] = asyncio.Queue(1)
+        messages_from_app: asyncio.Queue[Dict[str, Any]] = asyncio.Queue(1)
         messages_to_app: asyncio.Queue[Dict[str, Any]] = asyncio.Queue(1)
 
         async def disconnect_app():
             if scope["type"] == "http":
-                await messages_to_app.put(
-                    {"type": "http.disconnect"},
-                )
+                await messages_to_app.put({"type": "http.disconnect"})
             elif scope["type"] == "websocket":
-                await messages_to_app.put(
-                    {"type": "websocket.disconnect"},
-                )
+                await messages_to_app.put({"type": "websocket.disconnect"})
 
         async def handle_first_input_timeout():
             if scope["type"] == "http":
-                # TODO: prevent this from being sent if a response was already sent by the asgi app
-                #       also, prevent the (http) asgi app from sending additional data at this point
-                #       since this queues a full response
```

Comment:
We have handling in `modal-http` to make sure that responses are handled properly. The `ResponseDriver` state machine in `modal-http/src/service.rs` handles this.